openedx/features/badging/helpers.py

```diff
@@ -1,52 +1,80 @@
+import logging
+
 from collections import OrderedDict
 
+from lms.djangoapps.courseware.courses import get_course_by_id
+from lms.djangoapps.teams import is_feature_enabled as is_teams_feature_enabled
+from lms.djangoapps.teams.models import CourseTeam
+from nodebb.constants import TEAM_PLAYER_ENTRY_INDEX
+from nodebb.models import TeamGroupChat
+from openedx.features.badging.constants import TEAM_PLAYER
+
+from .constants import BADGES_KEY, FILTER_BADGES_ERROR
 from .models import Badge
 
+log = logging.getLogger('edx.badging')
+
 
-def populate_trophycase(courses, earned_badges):
+def populate_trophycase(user, courses, earned_badges):
     """
     This method populate trophycase data into dictionary and returns it. This method has a hierarchy of
     enrolled courses, badge type, earned and yet to be earned badges with complete detail
     :param courses: Courses enrolled by user
     :param earned_badges: Badges earned by user
+    :param user: Current logged-in user
     :return: dictionary containing trophycase json
     """
     trophycase_dict = OrderedDict()
     badge_queryset = Badge.objects.all().order_by('threshold')
```

Comment:
Why are we fetching the queryset here and passing it to the function as an argument when we can just accessing it inside the function, where it is being used?

Reply:
We are doing this once in `populate_trophycase` because the function where it is passed is being called from loop, hence prevented multiple db calls

Reply:
Great call! :+1: